docs/compiler.md

```diff
@@ -1,57 +1,55 @@
+is to ensure that the _Codegen_ component is capable of processing the _Compiler
+Output_.
 
-## Compiler interface
+## _Compiler_ interface
 
-The Compiler operates on the Compiler Input proto, effectively enabling any
-number of Frontends to interface with the Compiler in a language agnostic
-manner.
+The _Compiler_ operates on the _Compiler Input proto_ - enabling, any _Frontend_
+to interface with the _Compiler_ in a language agnostic manner.
 
-Similarly, the Compiler Output is also a proto that is then consumed by Codegen
-modules that can be written in any programming environment capable of
-communicating using Google Protobufs.
+Similarly, the _Compiler Output_ is also a _proto_ that is then consumed by
+_Codegen_ modules, able to be written in any programming environment capable of
+communicating via Google Protocol Buffers.
 
 ## Checking type definitions
 
-One primary purpose of the compiler is to typecheck the schemata that users
-define via the Frontend. The lambda calculus representation of schemata is "one
-level up" from the term level: ADT schemata declarations are the "terms" and
-kinds are the "types".
-
-Because the schema language only supports data declarations for algebraic data
-types, and does not support function definitions, the only two admissible kinds
-are `Type` and `TyCon`s of a specific arity, such that a `TyCon` can accept an
-arbitrary number of arguments of kind `Type` but can **only** accept arguments
-of kind `Type`. Put another way: An expression's kind is the same thing as its
-arity, and only expressions of arity=0 (i.e. expressions of kind `Type`) can be
-applied.
-
-To elaborate on the previous point: A user can define a higher kinded data type,
-such as `data Maybe a = Nothing | Just a` in the schema language. However users
-cannot define a data type which has a higher kinded type as a _parameter_, and
-users must "fully saturate" higher kinded types with well-scoped type variables
-or zero arity `Type`s to pass them as arguments to a `TyCon`. Consequently, a
-definition such as `data HKT f a = HKT (f a)` is not valid because `f :: Type ->
-Type`. Restricting the schema language in this way does reduce its
-expressiveness, but allowing higher kinded type parameters would greatly limit
-(or, at the very least, greatly complicate) our ability to codegen for a variety
-of target languages with type systems less expressive than Haskell.
-
-The schema language supports recursive types, and so the compiler must be
-capable of checking recursive types.
-
-In addition to typechecking, the compiler must perform a special check for
+The primary purpose of the compiler is to typecheck the schemata that users
+define via the _Frontend_. The schemata terms are ADT declarations and their
+types are kinds.
+
+Currently the schema language supports:
+
+ 1. types (such as `Int` or `Bool`),
+
+ 2. and type functions (such as `Maybe` or `Either`).
```

Comment:
I'd rather see you talk in terms of Kinds, and say things like,

Currently, the Compiler **accepts** type terms of kind `Type` and `Type -> Type` such as:
1. `Int :: Type`,
2. `Bool :: Type`,
3. `Maybe :: Type -> Type`,
4. `Either :: Type -> Type -> Type`.

All **type variables** are of kind `Type` as well

Reply:
I mention the all type variables later on.